File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/shared/config/MessageSourceConfig.java
package org.bootstrapbugz.api.shared.config;

import org.springframework.context.MessageSource;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.ReloadableResourceBundleMessageSource;

@Configuration
public class MessageSourceConfig {
  @Bean
  public MessageSource messageSource() {
    final var messageSource = new ReloadableResourceBundleMessageSource();
    messageSource.setBasename("classpath:/error-codes");
    messageSource.setDefaultEncoding("UTF-8");
    return messageSource;
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/config/MessageSourceConfig.kt
package org.bootstrapbugz.api.shared.config

import org.springframework.context.MessageSource
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.support.ReloadableResourceBundleMessageSource

@Configuration
class MessageSourceConfig {
  @Bean
  fun messageSource(): MessageSource {
    val messageSource = ReloadableResourceBundleMessageSource()
    messageSource.setBasename("classpath:/ValidationMessages")
    messageSource.setDefaultEncoding("UTF-8")
    return messageSource
  }
}
